Classes/Emogrifier.php

```diff
@@ -530,11 +536,21 @@ private function fillStyleAttributesWithMergedStyles()
     private function generateStyleStringFromDeclarationsArrays(array $oldStyles, array $newStyles)
     {
         $combinedStyles = array_merge($oldStyles, $newStyles);
+
+        $cacheKey = serialize($combinedStyles);
```

Comment:
As far as I know, serialize is quite slow. Have you checked whether json_encode or var_export on $combinedStyles might be faster?